Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIBookmarksCocoa.mm

```
@@ -47,7 +88,27 @@
 
 void WebExtensionAPIBookmarks::getRecent(long long numberOfItems, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
 {
-    callback->reportError(@"unimplemented");
+    if (numberOfItems < 1) {
+        NSString *errorMessage = @"Invalid argument: 'numberOfItems' must be at least 1.";
+        if (outExceptionString)
+            *outExceptionString = errorMessage;
+        callback->reportError(errorMessage);
```

You should match Chrome/Firefox here. You either want to throw an exception or `reportError()`, not both.

Also use `toErrorString()` for the error message. See other APIs for how it is used.